test cases/common/105 generatorcustom/meson.build

```diff
@@ -26,3 +26,20 @@ allinone = custom_target('alltogether',
 proggie = executable('proggie', 'main.c', allinone)
 
 test('proggie', proggie)
+
+# specifically testing that cross binaries are run with an exe_wrapper
+if meson.can_run_host_binaries()
+  add_languages('cpp', native : false)
+  gen_tool = executable('generator', 'gen.cpp', native : false)
```

Comment:
Aside: is there a particular reason to search for two compilers instead of one, in this test case? Writing a single line to a file and then closing it feels like something that should be easy enough to do in C... and then we use the resulting header in a C test program anyway.

Reply:
Honestly, I just hate writing C, its a miserable language which is rarely portable and so prone to simple programmer error that I'd rather not use it, especially in cases where I'm writing simple little programs and the C++ stdlib provides all the functionality I need without all of the corner cases to solve.

Reply:
It just feels meh because repeating that just means a bunch of additional fork+exec to detect and sanity check each compiler, and that's hardly of benefit to our long CI times.

Reply:
Which is the reason that I called `add_languages` inside the `if` block instead of adding them to the default language list, but I can use C if you prefer

Reply:
Okay, I rewrote it in C. I gave up on error handling half way through because C